### PR TITLE
eval: Stop executing *eval() function at error

### LINKED ISSUE #4822: pyeval() & py3eval() crashing
- Neovim version: NVIM v0.1.5-296-gbfbc974 
- [x] Vim behaves differently? Does not crash
- Operating system/version: OSX likely all others
- Terminal name/version: any

### Actual behaviour
When passing an unquoted argument to `pyeval` or `py3eval` the application will crash with an `EXC_BAD_ACCESS`. An invalid address is passed to `strlen` as the string causing it to crash when trying to dereference it (see lldb section below). 

When looking at the code, it appears that both `pyeval` and `py3eval` call `script_host_eval`. This function just assumes that the value passed to it is a valid string. 
```
list_append_string(args, argvars[0].vval.v_string, -1);
```
### Expected behaviour
Provide an error. For something like `:echo pyeval(1+1)` vim returns `2`. However, `:echo pyeval(sum(range(10)))` in vim returns an error.

### Steps to reproduce using `nvim -u NORC`
```
nvim -u NORC --cmd ':call pyeval(1+1)'

```

### Debug ouptut
```lldb
* thread #1: tid = 0xa2a8f, 0x00007fff99e22152 libsystem_c.dylib`strlen + 18, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x100000000)
    frame #0: 0x00007fff99e22152 libsystem_c.dylib`strlen + 18
libsystem_c.dylib`strlen:
->  0x7fff99e22152 <+18>: pcmpeqb xmm0, xmmword ptr [rdi]
    0x7fff99e22156 <+22>: pmovmskb esi, xmm0
    0x7fff99e2215a <+26>: and    rcx, 0xf
    0x7fff99e2215e <+30>: or     rax, -0x1

(lldb) re r rdi
     rdi = 0x0000000100000000

(lldb) bt
* thread #1: tid = 0xa2a8f, 0x00007fff99e22152 libsystem_c.dylib`strlen + 18, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x100000000)
    frame #0: 0x00007fff99e22152 libsystem_c.dylib`strlen + 18
    frame #1: 0x000000010f323f70 nvim`xstrdup(str="") + 32 at memory.c:378
  * frame #2: 0x000000010f236011 nvim`list_append_string(l=0x000000000f826dc0, str="", len=-1) + 97 at eval.c:5334
    frame #3: 0x000000010f261f86 nvim`script_host_eval(name="python", argvars=0x00007fff50a1da00, rettv=0x00007fff50a1dc18) + 134 at eval.c:21986
    frame #4: 0x000000010f253c17 nvim`f_pyeval(argvars=0x00007fff50a1da00, rettv=0x00007fff50a1dc18) + 39 at eval.c:12953
```

### PR DESCRIPTION
**Pull request by ZyX-I**
Fixes #4822
Closes #4823

### CODE REVIEW

File: test/functional/provider/python_spec.lua [hunk 1]
```diff
@@ -15,49 +25,58 @@ do
   end
 end
 
-describe('python commands and functions', function()
-  before_each(function()
-    clear()
-    command('python import vim')
-  end)
+before_each(function()
```
Comment (justinmk) [inline test/functional/provider/python_spec.lua]: Shouldn't this be in a describe block?

Comment (ZyX-I) [inline test/functional/provider/python_spec.lua]: @justinmk Why it should?

Comment (justinmk) [inline test/functional/provider/python_spec.lua]: busted docs only show examples inside `describe()`. Just wanted to be sure.

Anyways, LGTM.

Comment (ZyX-I) [inline test/functional/provider/python_spec.lua]: @justinmk Busted does not have thorough documentation. I know that this works locally and I know that this code works on travis and docs do not say that `before_each` *must not* be used like this (as well as that `before_each` *must* be placed inside `describe`), so I have this code. There will be more: a071e3896ecc7e2ff36fbcdbdc39e540bbca8511 (given issues with tests on branch this commit belongs to I guess I will move it to a separate PR).

**Pull request merged**